controllers/pxc/transition_key_copier.go

```diff
@@ -0,0 +1,134 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/go-logr/logr"
+
+	pxcv1 "github.com/Percona-Lab/k8s-vault-issuer/apis/pxc/v1"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (r *PerconaXtraDBClusterReconciler) processTransferVaultKeys(currClusterCR *pxcv1.PerconaXtraDBCluster, clustersStr string, log logr.Logger) error {
+	currClusterName := currClusterCR.Name + "." + currClusterCR.Namespace
+
+	clusters := strings.Split(clustersStr, ",")
+	failedClusters := make([]string, 0)
+
+	for _, v := range clusters {
+		splittedName := strings.Split(v, ".")
+		if len(splittedName) != 2 {
+			log.Error(nil, "invalid source cluster name, please use format clusterName.namespace", "cluster name", v)
+			failedClusters = append(failedClusters, v)
+			continue
+		}
+
+		clusterName, clusterNamespace := splittedName[0], splittedName[1]
+
+		srcClusterCR := &pxcv1.PerconaXtraDBCluster{}
+		err := r.Client.Get(context.TODO(),
+			types.NamespacedName{
+				Namespace: clusterNamespace,
+				Name:      clusterName,
+			}, srcClusterCR)
+		if err != nil {
+			log.Error(err, "get cluster srcClusterCR from namespace", "namespace", clusterNamespace)
+			failedClusters = append(failedClusters, v)
+			continue
+		}
+
+		values, ok := srcClusterCR.Annotations["percona.com/allow-transition-key-transfer"]
+		if !ok {
+			log.Error(nil, "no percona.com/allow-transition-key-transfer annotation found at cluster", "cluster name", v)
+			failedClusters = append(failedClusters, v)
+			continue
```

Comment:
To get rig of all of this `continue` calls you can move code to separate function and return error from it. And inside for loop just check it.

Reply:
8e25d8a